tests/vet/get-owner-contact-details.test.ts

```diff
@@ -0,0 +1,83 @@
+import { Pool } from "pg";
+import { withDb } from "../_db_helpers";
+import {
+  getVetActor,
+  insertDog,
+  insertUser,
+  insertVet,
+  userPii,
+} from "../_fixtures";
+import { dbInsertDogVetPreference } from "@/lib/data/db-dogs";
+import { OwnerContactDetails } from "@/lib/vet/vet-models";
+import { getOwnerContactDetails } from "@/lib/vet/actions/get-owner-contact-details";
+
+describe("getOwnerContactDetails", () => {
+  it("should return a dog's owner contact details", async () => {
+    await withDb(async (dbPool) => {
+      const { vetId } = await insertVet(1, dbPool);
+      const { dogId, ownerContactDetails } = await insertOwner(1, dbPool, {
+        preferredVetId: vetId,
+      });
+      const actor = getVetActor(vetId, dbPool);
+      const { result, error } = await getOwnerContactDetails(actor, dogId);
+      expect(error).toBeUndefined();
+      expect(result).toEqual(ownerContactDetails);
+    });
+  });
+  it("should return ERROR_UNAUTHORIZED when vet is not a preferred vet", async () => {
```

Comment:
Test this case also...

- should return ERROR_UNAUTHORIZED when vet is a preferred vet for the wrong dog
```
GIVEN vets v1 and v2
AND user with two dogs d1 and d2
AND the preferred vet for d1 is v1
AND the preferred vet for d2 is v2
WHEN v1 requests for owner details of d2
THEN expect ERROR_UNAUTHORIZED
```

Reply:
Added test in https://github.com/barkbank/barkbank-pawtal/pull/115/commits/be9de65b06c11017200242ff3f3f77938f4db60a